best-practices/index-best-practices.md

```diff
@@ -0,0 +1,325 @@
+
+- Unused indexes: these indexes are once relevant but are no longer selected by the query optimizer, consuming storage and adding unnecessary overhead to write operations.
+- Inefficient indexes: some indexes are used by the optimizer but scan more data than expected, increasing disk I/O and slowing down query performance.
+
+If left unaddressed, these indexing issues can cause higher storage costs, degraded performance, and operational inefficiencies. In a distributed SQL database like TiDB, indexing inefficiencies have an even greater impact due to the scale of distributed queries and the complexity of multi-node coordination. That is why regular index audits are crucial for maintaining an optimized database. 
+
+Proactively identifying and optimizing indexes helps:
+
+- Reduce storage overhead: removing unused indexes frees up disk space and reduces long-term storage costs.
+- Improve write performance: write-heavy workloads (such as `INSERT`, `UPDATE`, and `DELETE`) perform better when unnecessary index maintenance is eliminated.
+- Optimize query execution: efficient indexes reduce the number of rows scanned, improving query speed and response times.
+- Streamline database management: fewer and well-optimized indexes simplify backups, recovery, and schema changes.
+
+TiDB v8.0.0 introduces the [`TIDB_INDEX_USAGE`](/information-schema/information-schema-tidb-index-usage.md) table and the [`schema_unused_indexes`](/sys-schema/sys-schema-unused-indexes.md) table to help you track index usage patterns and make data-driven decisions. 
+
+Because indexes evolve with changing business logic, regular index audits are a standard part of database maintenance. TiDB provides built-in observability tools to help you detect, evaluate, and optimize indexes without risk.
+
+This document describes the tools that you can use to detect and eliminate unused or inefficient indexes, thus improving TiDB's performance and stability.
+
+## TiDB index optimization: a data-driven approach
+
+Indexes are essential for query performance, but removing them without proper analysis can lead to unexpected regressions or even system instability. To ensure safe and effective index management, TiDB provides built-in observability tools that allow you to:
+
+- Track index usage in real-time: identify how often an index is accessed and whether it contributes to performance improvements.
+- Detect unused indexes: locate indexes that have not been used since the database is last restarted.
+- Assess index efficiency: evaluate whether an index filters data effectively or causes excessive I/O overhead.
+- Safely test index removal: temporarily make an index invisible before deleting it to ensure no queries depend on it.
+
+TiDB simplifies index optimization by introducing the following powerful tools:
+
+- `TIDB_INDEX_USAGE`: monitors index usage patterns and query frequency.
+- `schema_unused_indexes`: lists indexes that have not been used since the database is last restarted.
```

Comment:
```suggestion
- `INFORMATION_SCHEMA`.`TIDB_INDEX_USAGE`: monitors index usage patterns and query frequency.
- `mysql`.`schema_unused_indexes`: lists indexes that have not been used since the database is last restarted.
```